### DISCUSSION TITLE
Compatibility with whitespace handling in Jekyll: trim file on {% include %}

### DISCUSSION THREAD

**Discussion by Nowaker** [harttle/liquidjs#521] [Working with Jekyll Sites]
On `{% include blah.html %}`, Jekyll start-trims (removes leading whitespace) from the referenced file - equivalent of `fs.fileReadSync('blah.html').trimStart()` - before embedding it into the output.

I'd like to ask for a new option(s) for https://liquidjs.com/tutorials/whitespace-control.html#via-Options:

- `trimIncludeLeft` & `trimIncludeRight`
- `trimLayoutLeft`  & `trimLayoutRight`- note: Jekyll uses `layout: blah` in front matter for layouts, so I never used `{% layout %}` in my project, so I will not use it, but it may make sense to include it for consistency with `trimIncludes`

Because `{% include %}` is a built-in tag, I wasn't able to tune it to ensure Jekyll compatibility.

FYI: My Jekyll-in-Gatsby project is now almost complete, with whitespace being the only difference between the builds.

**Comment by harttle** [comment]
Hey @Nowaker , we can treat this as a bugfix for [Jekyll include](https://liquidjs.com/tutorials/options.html#Jekyll-include). As for `layout`, I don't think it makes as much sense as `include`:

- it's not related to jekyll compatibility, and
- I can't think of a scenario to trim a parent template (or root template), it's essentially the same funtionality as trimming current template when calling `renderFile()`

**Comment by Nowaker** [comment]
> we can treat this as a bugfix for Jekyll include

That's a fantastic idea!

**Comment by harttle** [comment]
I created a demo for this on Jekyll: https://github.com/harttle/jekyll-include-demo The [_layouts/foo.liquid](https://github.com/harttle/jekyll-include-demo/blob/master/_layouts/foo.liquid) included [_includes/b.liquid](https://github.com/harttle/jekyll-include-demo/blob/master/_includes/b.liquid) which has leading & trailing spaces and new lines.

It turns out, in this demo, the leading & trailing spaces and new lines of the included file are preserved:

![image](https://user-images.githubusercontent.com/4427974/163723180-bdb6412e-e081-4e05-850d-fe86788f1fe5.png)
